src/backends/torch/torchinputconns.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * DeepDetect
+ * Copyright (c) 2018 Jolibrain
+ * Author: Louis Jean <ljean@etud.insa-toulouse.fr>
+ *
+ * This file is part of deepdetect.
+ *
+ * deepdetect is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * deepdetect is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with deepdetect.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TORCHINPUTCONNS_H
+#define TORCHINPUTCONNS_H
+
+#include "imginputfileconn.h"
+#include "txtinputfileconn.h"
+
+#include <vector>
+
+namespace dd
+{
+    class TorchInputInterface
+    {
+    public:
+      TorchInputInterface() {}
+      TorchInputInterface(const TorchInputInterface &i)
+      {
+
+      }
+
+      ~TorchInputInterface() {}
+    };
+
+    class ImgTorchInputFileConn : public ImgInputFileConn, public TorchInputInterface
+    {
+    public:
+        ImgTorchInputFileConn()
+            :ImgInputFileConn() {}
+        ImgTorchInputFileConn(const ImgTorchInputFileConn &i)
+            :ImgInputFileConn(i),TorchInputInterface(i) {}
+        ~ImgTorchInputFileConn() {}
+
+        // for API info only
+        int width() const
+        {
+            return _width;
+        }
+
+        // for API info only
+        int height() const
+        {
+            return _height;
+        }
+
+        void init(const APIData &ad)
+        {
+            ImgInputFileConn::init(ad);
+        }
+        
+        void transform(const APIData &ad)
+        {
+            try
+            {
+                ImgInputFileConn::transform(ad);
+            }
+            catch(const std::exception& e)
+            {
+                throw;
+            }
+
+            std::vector<at::Tensor> tensors;
+
+            for (const cv::Mat &bgr : this->_images) {
+                _height = bgr.rows;
+                _width = bgr.cols;
+
+                std::vector<int64_t> sizes{ _height, _width, 3 };
+                at::TensorOptions options(at::ScalarType::Byte);
+                at::Tensor imgt = torch::from_blob(bgr.data, at::IntList(sizes), options);
+                imgt = imgt.toType(at::kFloat).mul(1./255.).permute({2, 0, 1});
```

Comment:
Let's keep the normalization and channel permutation from BGR here, but let's add an option that would default to `true` for now. The reason is that once we start using models that are trained via DD and libtorch, normalization and especially permutation may not bee needed anymore.

With other backends, we typically use `std` in input connector to specify the normalization factor (here 255). 

DD uses OpenCV and thus is BGR. Instead of permuting, we may want to add an `rgb` boolean API parameter to the input connector, and let our main `ImgInputConn` do the conversion ?